routes/freelance/router.js

```diff
@@ -236,13 +237,49 @@ router.put('/:freelanceid/availability', function(req, res, next) {
 
 // POST freelance/
 router.post('/', function(req, res, next) {
-  const newFreelance = new Freelance(req.body);
+  if (req.body.tags != undefined) {
+    var tags = req.body.tags.split(',');
+  }
+  
+  let parameters = req.body;
+  delete parameters.tags;
+  const newFreelance = new Freelance(parameters);
+  // Save Freelance to get its id.
   newFreelance.save(function(err, saved) {
     if (err) {
       res.status(400).json(utils.formatErrorMessage(err));
+      return;
+    }
+    // Now we can parse, create and save the tags.
+    let tagObjects = [];
+    let count = 0;
+    if (tags != undefined) {
+      for (let tag of tags) {
```

Comment:
What if one or more of the tags that the user entered already exists? If I understood correctly this method doesn't check that.